crates/rpc/src/v03/method/get_events.rs

```diff
@@ -0,0 +1,893 @@
+use crate::context::RpcContext;
+use anyhow::Context;
+use pathfinder_common::{BlockId, ContractAddress, EventKey, StarknetBlockNumber};
+use pathfinder_storage::{
+    EventFilterError, StarknetBlocksTable, StarknetEventFilter, StarknetEventsTable, V03KeyFilter,
+};
+use serde::Deserialize;
+use starknet_gateway_types::pending::PendingData;
+use tokio::task::JoinHandle;
+
+// FIXME: allow `generate_rpc_error_subset!` to work with enum struct variants. This may not actually be possible though.
```

Comment:
I don't think its generally possible - I've tried a few times.

However, maybe one can make it very specific - i.e. have a separate arm for `TooManyKeysInFilter` exactly.. and precreate its conditions. Not sure if its worth it though, considering this will only get used once.

Reply:
Definitely not worth that IMO. I think I'll just remove the FIXME...